scenariolib/uabot_test.go

```diff
@@ -43,3 +47,53 @@ func equals(tb testing.TB, exp, act interface{}) {
 		tb.FailNow()
 	}
 }
+
+type ExpectedRequest struct {
+	Method  string
+	Headers map[string]string
+	Body    []byte
+}
+
+// JSONBytesEqual compares the JSON in two byte slices.
+func JSONBytesEqual(a, b []byte) (bool, error) {
+	var j, j2 interface{}
+	if err := json.Unmarshal(a, &j); err != nil {
+		return false, err
+	}
+	if err := json.Unmarshal(b, &j2); err != nil {
+		return false, err
+	}
+	return reflect.DeepEqual(j2, j), nil
+}
+
+func createMockServer(t testing.TB, expected map[string]ExpectedRequest) *httptest.Server {
+	// Start a local HTTP server to intercept requests
+	// Using url to match the expected responses above.
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		url := req.URL.String()
+
+		expReq, exists := expected[url]
+		assert(t, exists, "MISSING expected request for %s", url)
```

Comment:
I would not assert here, I would return an empty response or an error. 
IMO it's not the mock's job to test if a mock response was set.

The test should assert false because it didn't get the response it was expecting, not the server saying it didn't get a mock response.

Reply:
Can we talk about this? 
Not sure what you mean. 

It's not just a mock, more like a proxy. I see I used a bad name for it. 
I think the asserts have to be in the request _handler_.